docs/content/docs/dev/datastream/fault-tolerance/checkpointing.md

```diff
@@ -292,4 +292,25 @@ The final checkpoint would be triggered immediately after all operators have rea
 without waiting for periodic triggering, but the job will need to wait for this final checkpoint 
 to be completed.
 
+## Unify file merging mechanism for checkpoints
+
+Flink 1.20 introduces a unified file merging mechanism for checkpointing, which allows scattered small checkpoint files 
+to be written into a single file, reducing the number of file creations and file deletions, helping to alleviate the pressure
+of file system metadata management and file flooding problem. The unified fie merging mechanism can be enabled by setting
+the property `state.checkpoints.file-merging.enabled` to `true`. **Note** that enabling this mechanism may lead to space amplification,
+that is, the actual occupation on the file system will be larger than the checkpoint size. `state.checkpoints.file-merging.max-space-amplification` 
+can be used to limit the upper bound of space amplification.
+
+This mechanism is applicable to all types of state files in Flink, including keyed state, operator state, channel state 
+and changelog DSTL files. Subtask level granular merging is provided for shared scope state; TaskManager-level granular merging 
```

Comment:
some unsupported types, e.g. DSTL files could be removed for now or listed as a limitation ?